x/programs/rust/examples/csamm/src/lib.rs

```diff
@@ -0,0 +1,169 @@
+use std::cmp::Ordering;
+use wasmlanche_sdk::{params, program::Program, public, state_keys, types::Address};
+
+// Implementation of a CSAMM. This assumes a 1:1 peg between the two assets or one of them is going to be missing from the pool.
+
+/// The program state keys.
+#[state_keys]
+enum StateKeys {
+    // TODO this is also an LP token, could we reuse parts of the "token" program ?
+    // TotalSupply,
+    // Balance(Address),
+    /// The vec of coins. Key prefix 0x0 + addresses
```

Comment:
If I'm understanding your question correctly, you could just create a single key called "Coins" and put an array as the value?

Reply:
Absolutely, that works, not sure if it's worth generalizing the amount of tokens, that would be pretty cool

Reply:
It is currently not possible to add a `Coins(Program)` key simply because `Program` doesn't implement `Debug`.
Would it make sense for it to implement it, or should we make the user pass the array of programs and check if the program `id` is matching the one in the storage everytime ?